File: back/src/main/java/br/com/pipo/model/Employee.java
package br.com.pipo.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Data
@Document
public class Employee implements Serializable {

    @Id
    private String id;

    private String cpf;

    private String name;

    private LocalDate admissionDate;

    private String email;

    private String weight;

    private String height;

    private BigDecimal meditationHours;

    @DBRef
    private Client client;

    @DBRef
    private List<Product> products;
}

File: front/model/Employee.ts
import { Address } from "./Address";
import { Client } from "./Client";
import { Product } from "./Product";

export interface Employee {
  id?: string;
  name?: string;
  cpf?: string;
  email?: string;
  admissionDate?: Date;
  weight?: string;
  height?: string;
  client?: Client;
  meditationHours?: Number;
  products?: Product[];
  address?: Address;
}